tests/utils/test_common.py

```diff
@@ -159,38 +178,123 @@ def test_cli_missing_log_level_default_used():
 def test_cli_missing_log_level_env_var_used():
     """Test CLI without log level uses env var for both rasa and libraries."""
     configure_logging_and_warnings()
     rasa_logger = logging.getLogger("rasa")
-    rasa_logger.level == logging.WARNING
+    assert rasa_logger.level == logging.WARNING
     matplotlib_logger = logging.getLogger("matplotlib")
-    matplotlib_logger.level == logging.INFO
+    assert matplotlib_logger.level == logging.INFO
+
+
+def test_cli_valid_logging_configuration() -> None:
+    logging_config_file = "data/test_logging_config_files/test_logging_config.yml"
+    configure_logging_from_file(logging_config_file=logging_config_file)
+    rasa_logger = logging.getLogger("rasa")
+
+    handlers = rasa_logger.handlers
+    assert len(handlers) == 1
+    assert isinstance(handlers[0], logging.FileHandler)
+    assert "test_handler" == rasa_logger.handlers[0].name
+
+    logging_message = "This is a test info log."
+    rasa_logger.info(logging_message)
+
+    handler_filename = handlers[0].baseFilename
+    assert Path(handler_filename).exists()
+
+    with open(handler_filename, "r") as logs:
+        data = logs.readlines()
+        logs_dict = json.loads(data[-1])
+        assert logs_dict.get("message") == logging_message
+
+        for key in ["time", "name", "levelname"]:
+            assert key in logs_dict.keys()
+
+    # reset logging config
+    reset_logging()
+
+
+@pytest.mark.parametrize(
+    "logging_config_file",
+    [
+        "data/test_logging_config_files/test_missing_required_key_invalid_config.yml",
+        "data/test_logging_config_files/test_invalid_value_for_level_in_config.yml",
+        "data/test_logging_config_files/test_invalid_handler_key_in_config.yml",
+    ],
+)
+def test_cli_invalid_logging_configuration(
+    logging_config_file: Text, caplog: LogCaptureFixture
+) -> None:
+    with caplog.at_level(logging.DEBUG):
+        configure_logging_from_file(logging_config_file=logging_config_file)
+
+    assert (
+        f"The logging config file {logging_config_file} could not be applied "
+        f"because it failed validation against the built-in Python "
+        f"logging schema." in caplog.text
+    )
+
+
+@pytest.mark.skipif(
+    sys.version_info.minor == 7, reason="no error is raised with python 3.7"
+)
+def test_cli_invalid_format_value_in_config(caplog: LogCaptureFixture) -> None:
+    logging_config_file = (
+        "data/test_logging_config_files/test_invalid_format_value_in_config.yml"
+    )
+
+    with caplog.at_level(logging.DEBUG):
+        configure_logging_from_file(logging_config_file=logging_config_file)
+
+    assert (
+        f"The logging config file {logging_config_file} could not be applied "
+        f"because it failed validation against the built-in Python "
+        f"logging schema." in caplog.text
+    )
+
+
+@pytest.mark.skipif(
+    sys.version_info.minor == 9, reason="no error is raised with python 3.9"
+)
```

Comment:
same as above, odd 🤔

Reply:
Yeah, this particular config doesn't raise any error in logging module in Python 3.9.
I can go and look through the changelogs from python if you want?

Reply:
I didn't find anything. But I'm guessing that we should see an error in the logs if the handler is not set properly. So maybe the behaviour changed from 3.8 to 3.9, and the validation is not done at the same time. I think it's safe to keep it that way for now